library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -696,9 +704,83 @@ public boolean isLoading() {
 
   @Override
   public void reevaluateBuffer(long positionUs) {
-    // Do nothing.
+    if (hlsInterruptStateDelegate != null && REQUESTED.equals(hlsInterruptStateDelegate.getCancelState()) && loader.isLoading()) {
+      hlsInterruptStateDelegate.setCancelState(TRIGGERED);
```

Comment:
Related to my other comment about not needing this delegate:
You can query chunkSource.getPreferredQueueSize as below. And only if you need to discard samples, then you can check if you are currently loading. If yes, cancel the load and set your isInterrupted flag. If no, discard immediately.